### PR TITLE
tcp: fix overenthusiastic bounds on the new pool

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: Low
Testing: new integration test
Docs Changes: made API more clear when requests count as connections :-/
Release Notes: inline
Fixes https://github.com/envoyproxy/envoy/issues/19033

### CODE REVIEW

File: api/envoy/config/cluster/v3/circuit_breaker.proto [hunk 1]
```diff
@@ -59,10 +59,12 @@ message CircuitBreakers {
 
     // The maximum number of pending requests that Envoy will allow to the
     // upstream cluster. If not specified, the default is 1024.
+    // This limit is applied as a connection limit for non-HTTP traffic.
     google.protobuf.UInt32Value max_pending_requests = 3;
 
     // The maximum number of parallel requests that Envoy will make to the
     // upstream cluster. If not specified, the default is 1024.
+    // This limit is not applied as a connection limit for non-HTTP traffic.
```
Comment (ggreenway) [inline api/envoy/config/cluster/v3/circuit_breaker.proto:67]: "This limit does not apply for non-HTTP traffic"?

**Review by ggreenway — CHANGES_REQUESTED** [review]

**Review by ggreenway — DISMISSED** [review]
LGTM

**Comment by ggreenway** [comment]
This should probably be backported.

**Review by ggreenway — APPROVED** [review]

**Review by ggreenway — APPROVED** [review]

**Review by adisuissa — COMMENTED** [review]
/lgtm api

**Pull request merged**